tuner/tuner/common.py

```diff
@@ -38,10 +39,20 @@ def getI64(self, value: int) -> ir.IntegerAttr:
 
 
 class TunerContext:
-    def __init__(self, mlir_ctx: ir.Context, logger: logging.Logger):
-        self.mlir_ctx: ir.Context = mlir_ctx
-        self.logger: logging.Logger = logger
-        self.type: CommonTypes = CommonTypes(mlir_ctx)
+    def __init__(self, logger: Optional[logging.Logger] = None):
+        self.mlir_ctx: ir.Context = ir.Context()
+        self.logger: logging.Logger = logger or logging.getLogger(
+            "tune"
+        )  # Default to "tune" logger
+        self.type: CommonTypes = CommonTypes(self.mlir_ctx)
+
+    def __enter__(self) -> TunerContext:
+        self.mlir_ctx.__enter__()
+        return self
+
+    def __exit__(self, exc_type, exc_value, traceback) -> Literal[False]:
```

Comment:
Can we add type hints for `exc_type`, `exc_value`, and `traceback`? Also, can we make this return whatever the nested exit returns?